### ISSUE TITLE
MacOS 14.7 -- scroll no longer works [bug]

### ISSUE THREAD

**Issue by CtrlAltID** [noah-nuebling/Mac-Mouse-Fix#1121]
## Description

Since the MacOS update, I have to turn off Smoothness in order for the mouse scroll wheel to function.

## Steps to reproduce

Scroll does not work when Smoothness is either Regular or High. Turning Smoothness Off enables the scroolwheel to work again, albite without a nice smooth feel. 

## Context

Mac Mouse Fix Version: 2.2.4
macOS Version: 14.7
Mouse Model: Logitech Performance MX

## Additional Comments

–

---

### Console Logs

–

### Crash Reports

–

**Comment by CooLeroff** [comment]
Same on MacOS 15.0, I turn off smooth scrolling for use mouse wheel

**Comment by noah-nuebling** [comment]
Thanks for the feedback!

Are you guys still experiencing this problem?

**Comment by CtrlAltID** [comment]
3.03 fixed my issue. Thank you for the quick and smooth response!Regards,ColeOn Sep 21, 2024, at 12:51 PM, Noah Nuebling ***@***.***> wrote:﻿
Thanks for the feedback!
Are you guys still experiencing this problem?

**Comment by ingolohr** [comment]
@noah-nuebling Yes, still experiencing the issue on Version 3.0.3 (22854). M1 MBP, macOS 14.7.

**Comment by noah-nuebling** [comment]
Hi @ingolohr, thank you for reaching out!

So based on the other reports I received about this problem, different users have reported that the issue went away after:

1. Disabling and then re-enabling Mac Mouse Fix
2. Rebooting the computer
3. Just using the computer for a while until the issue 'resolved itself'
4. Updating to a new macOS version while they were experiencing the issue
5. Updating to Mac Mouse Fix 3.0.3

Have you tried any of these steps? (Aside from 5. which you already mentioned) 
If so, have you noticed any difference before and after?

I recommend you try steps 1. and 2. if you haven't already. If that doesn't fix the issue, I will look into it further.

**Comment by ingolohr** [comment]
Many thanks for the quick reply. 

Yes, I have tried 1. and 2., and both "fix" the issue temporarily, until it comes back (usually after waking from sleep). A more permanent actual fix would be great. 

FWIW, I haven't run into the issue on my work MBP, an M2 Max, also macOS 14.7, yet.

**Comment by ingolohr** [comment]
@noah-nuebling Is this something you're still looking into? I basically have to do the dis/reenable dance almost every time I wake my MBP from sleep, which isn't great. Thanks!

**Comment by noah-nuebling** [comment]
Hey @ingolohr sorry for the late reply.

I'd definitely like to keep looking into this!

So first off, a few more basic questions:

1. Do you have any other software installed that might change your scrolling/mouse behaviour such as Logitech Options?
2. Are there any other differences between your M2 MacBook and your M1 MacBook that you think might have something to do with this?

With that out of the way, here's a **debug version** of Mac Mouse Fix 3.0.3: [Mac Mouse Fix.app.zip](https://github.com/user-attachments/files/17165522/Mac.Mouse.Fix.app.zip)

This debug version is a little slower than the regular version, but it logs its state to the console extensively.
If you can collect the logs that Mac Mouse Fix sends around the time that the issue occurs, that might give us the necessary insight to understand why the issue happens.

To collect logs:

1. Go to `Console.app > Action` and enable 'Include Info Messages' and 'Include Debug Messages'
2. Go to `Console.app > Search` and enter 'Mac Mouse Fix'
3. Go to `Console.app > [Your device name]` and click 'Start streaming'
4. Reproduce the bug using the **debug version** of Mac Mouse Fix which I attached above
5. Copy-paste logs which occured around the time of the bug and send them to me
    Notes: 
    - There are going to be hundreds or thousands of logs per minute, and you might want to send a few minutes of logs if you're not exactly sure when the bug occured. 
    - You might want to use https://pastebin.com/ or a similar site to send all that text.

I understand this is a bit tedious. I'm thinking of building a 'debug info recording mode' into Mac Mouse Fix to streamline this process. But for now, this is the best way I can think of to further investigate the bug.

Edit: Oh I forgot to mention:

You can also look for `Console.app > Crash Reports` and send me any reports you find that might be related.

**Comment by noah-nuebling** [comment]
Oh I forgot to mention:

You can also look for `Console.app > Crash Reports` and send me any reports you find that might be related.

**Comment by lorrin** [comment]
I'm a relatively new Mac Mouse Fix user; I think 3.0.3 is my first version. Did not experience this issue initially, but it started immediately after I upgraded from macOS 14.7.0 to 14.7.1. I am running the [Contour RollerMouse driver](https://contourdesign.com/pages/drivers) in addition to Mac Mouse Fix.

**Comment by ingolohr** [comment]
@noah-nuebling Update (sorry, didn't have time to use the debug version yet): This happens basically all the time on both my private M1 and work M2 Max MBP, always when waking from sleep. I'd day this happens about 7-8 out of 10 times.

**Comment by noah-nuebling** [comment]
Hey guys, if anybody is still experiencing the scrolling issue, and you would like to help look into it more deeply:

Just send me [an email](https://redirect.macmousefix.com/?target=mailto-noah), and I'll provide you with a debug build of Mac Mouse Fix that logs everything its doing in a detailed way. You'll be able to collect the logs with a special keyboard shortcut and upload them to my upload link.

I know that running Console.app all the time and copy-pasting the logs isn't great – this new process should be much easer!

I'd love to streamline the process of analyzing and debugging hard-to-reproduce issues like this one – and I'd love to solve this specific issue, and have time to look into it now – so I'd be really grateful for any feedback. 

@lorrin Thank you for the info! I did suspect that it might be related to interference with other mouse drivers, too, but I couldn't see such a pattern among the bug reports so far.

@ingolohr That does sound annoying. It's also interesting that it happens on both of your machines – I've never experienced this issue on my personal M1 machine – so perhaps there is something specific about your usage patterns that triggers this? Anyways, as I mentioned above, I'm sorry for not addressing the issue sooner. But I'm available now in case you're still interested in resolving it.

Also see: https://github.com/noah-nuebling/mac-mouse-fix/labels/Scroll%20Stops%20Working%20Intermediately

**Comment by noah-nuebling** [comment]
To anyone who is still experiencing the issue, here's an **update**:

You **don't have to send me an email** – I've streamlined the process, and you can now just follow the instructions here to **collect debug logs** which will be *highly valuable* for understanding and fixing the issue:

1. Install, and enable the special debug build of Mac Mouse Fix
    - Download link: https://mega.nz/file/So4wAa4I#HzWf0twuKj0o78qOTzecV5OIHPe1BC0ZTmEG6TxL76k
    - (This version records everything it's doing in detailed logs.)
    - (Make sure to set the scrolling smoothness to 'High' or 'Regular' to cause the bug to occur.)
2. When scrolling stops working
    - a) Take a **screenshot** (Shift-Command-3) to record the current time.
    - b) Document the problem in the logs
        - Try scrolling for a bit
        - Go to the 'Scrolling' tab in Mac Mouse Fix, then set the 'Smoothness' to 'High', then to 'Off', and then back to 'High'.
        - Try scrolling some more
        - Go to the 'General' tab in Mac Mouse Fix, then disable, and re-enable the app
        - Try scrolling *even* more
    - c) Collect the logs in a **sysdiagnose** file
        - Press Shift-Control-Option-Command-Period
            - The screen will flash white to show the sysdiagnose process has started.
        - Wait a bit (up to 10 minutes) for completion.
            - The sysdiagnose file will appear in Finder at `/private/var/tmp/sysdiagnose_[...].tar.gz`
3. Send me the files:
    - Please upload both the **screenshot** and the **sysdiagnose** file to this link: https://mega.nz/filerequest/L2KHMWEXKp4
        - After uploading the files, don't forget to copy the 'Upload ID' from the upload page, and send it to me (either by commenting here or [via email](https://redirect.macmousefix.com/?target=mailto-noah)). That way I'll know which files were uploaded by you.
    - (You can also send the files another way, but the sysdiagnose file is typically over 100 MB and too large to send directly via GitHub or Email.)
4. Disable logging
    - After sending the files, you can switch back to a regular release of Mac Mouse Fix to stop the detailed logging (which uses extra CPU).

Privacy Note:
The sysdiagnose file contains detailed information about your system.
Personal information is normally redacted, but I think some might remain if there are certain bugs in the software on your system.
In any case, I will not share your sysdiagnose file with anyone without your permission.

---

Thank you for helping to make the app more stable!

**Comment by noah-nuebling** [comment]
Hello everyone,

I addressed this issue in the latest [MMF 3.0.6](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/3.0.6).

Let me know if there are still any issues.
